nutkit/frontend/driver.py

```diff
@@ -59,29 +55,36 @@ def receive(self, timeout=None, hooks=None, *, allow_resolution):
                         hooks=hooks
                     )
                     continue
-            if self._bookmark_manager_config is not None:
-                supply = self._bookmark_manager_config.bookmarks_supplier
-                if supply is not None:
-                    if isinstance(res, protocol.BookmarksSupplierRequest):
+            if isinstance(res, protocol.BookmarksSupplierRequest):
+                if res.bookmarkManagerId in self._bookmarks_managers:
```

Comment:
Shouldn't we fail hard if the backend requests a callback on an undefined BMM?

Reply:
It will fail, the res will be return and the receiver will get an expected object and fail. This is the behaviour with the resolution messages. We could raise for being more explicit.